CONTRIBUTING.rst

```diff
@@ -120,34 +141,74 @@ On OS X, Homebrew is a reliable way to get GDAL.
 Python build requirements
 -------------------------
 
-Provision your virtualenv with Rasterio's build requirements. Rasterio's
-setup.py script will not run unless Cython and Numpy are installed, so do this
-first from the Rasterio repo directory.
+Provision a virtualenv with Rasterio's build requirements.  Rasterio's
+``setup.py`` script will not run unless Cython and Numpy are installed, so do
+this first from the Rasterio repo directory.
+
+Linux users may need to install some additional Numpy dependencies:
 
 .. code-block:: console
 
-    (riodev)$ pip install -U pip
-    (riodev)$ pip install -r requirements-dev.txt
+    $ sudo apt-get install libatlas-dev libatlas-base-dev gfortran
+
+then:
+
+.. code-block:: console
+
+    $ pip install -U pip
+    $ pip install -r requirements-dev.txt
 
 Installing Rasterio
 -------------------
 
-Once that's done, install Rasterio in editable mode with the "docs,test"
-extras.
+Rasterio, its Cython extensions, normal dependencies, and dev dependencies can
+be installed with ``$ pip``.  Installing Rasterio in editable mode while
+developing is very but only affects the Python files.  Specifying the
+``[test]`` extra in the command below tells ``$ pip`` to also install
+Rasterio's dev dependencies.
+
+.. code-block:: console
+
+    $ pip install -e .[test]
+
+Any time a Cython (``.pyx`` or ``.pxd``) file is edited the extension modules
+need to be recompiled, which is most easily achieved with:
 
 .. code-block:: console
 
-    (riodev)$ pip install -e .[docs,test]
+    $ pip install -e .
+
+In some cases (when switching between Python 2 and 3?) an exception is raised
```

Comment:
@geowurster perhaps we should be more direct.  If you switch python versions, **always** recompile.  If you get this warning, it is a reminder to do that.